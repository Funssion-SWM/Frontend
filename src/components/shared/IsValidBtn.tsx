type Props = {
  text: string;
  onClick: () => void;
  isValid: boolean;
  disabled?: boolean;
};

export default function IsValidBtn({
  text,
  onClick,
  isValid,
  disabled = false,
}: Props) {
  return (
    <button
      type="button"
      className={`${
        isValid
          ? 'bg-soma-blue-20 text-soma-blue-50'
          : 'bg-soma-grey-30 text-soma-grey-45'
      } py-2 px-3.5 rounded-3xl mx-1 transition hover:bg-soma-blue-20 hover:text-soma-blue-50`}
      onClick={() => onClick()}
      disabled={disabled}
    >
      {text}
    </button>
  );
}
