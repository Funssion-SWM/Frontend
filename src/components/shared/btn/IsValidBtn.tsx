type Props = {
  text: string;
  onClick: () => void;
  isValid: boolean;
  disabled?: boolean;
};

export default function IsValidBtn({
  text,
  onClick,
  isValid,
  disabled = false,
}: Props) {
  return (
    <button
      type="button"
      className={`${
        isValid
          ? 'bg-soma-blue-20 text-soma-blue-50'
          : 'bg-soma-grey-30 text-soma-grey-45'
      } min-w-fit py-2 px-3.5 rounded-3xl mx-1 transition hover:bg-soma-blue-20 hover:text-soma-blue-50 text-sm sm:text-base prose`}
      onClick={() => onClick()}
      disabled={disabled}
    >
      {text}
    </button>
  );
}
