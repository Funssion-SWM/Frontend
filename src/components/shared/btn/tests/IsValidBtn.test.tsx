import { render, screen } from '@testing-library/react';
import IsValidBtn from '../IsValidBtn';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

describe('isValidBtn', () => {
  it('renders correctly when isValid is true', () => {
    const component = renderer.create(
      <IsValidBtn text={'text'} isValid={true} onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when isValid is false', () => {
    const component = renderer.create(
      <IsValidBtn text={'text'} isValid={false} onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('when button clicked', async () => {
    const mockCallBack = jest.fn();

    render(<IsValidBtn text="text" isValid={true} onClick={mockCallBack} />);

    const btn = screen.getByRole('button');

    await userEvent.click(btn);
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
