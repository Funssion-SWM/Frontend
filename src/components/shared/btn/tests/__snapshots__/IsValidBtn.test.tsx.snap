// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isValidBtn renders correctly when isValid is false 1`] = `
<button
  className="bg-soma-grey-30 text-soma-grey-45 min-w-fit py-2 px-3.5 rounded-3xl mx-1 transition hover:bg-soma-blue-20 hover:text-soma-blue-50 text-sm sm:text-base"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  text
</button>
`;

exports[`isValidBtn renders correctly when isValid is true 1`] = `
<button
  className="bg-soma-blue-20 text-soma-blue-50 min-w-fit py-2 px-3.5 rounded-3xl mx-1 transition hover:bg-soma-blue-20 hover:text-soma-blue-50 text-sm sm:text-base"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  text
</button>
`;
